src/controller/python/chip-device-ctrl.py

```diff
@@ -635,11 +635,63 @@ def echo_alive(message):
     print(message)
     return message
 
-def ble_scan() -> Dict[Any, Any]:
+
+def resolve(fabric_id: int, node_id: int) -> Dict[str, Any]:
     try:
         __check_supported_os()
-        device_manager.do_blescan("")
+        err = device_manager.devCtrl.ResolveNode(fabric_id, node_id)
+        if err == 0:
+            address = device_manager.devCtrl.GetAddressAndPort(int(args[1]))
+            if address is not None:
+                address = "{}:{}".format(
+                    *address)  
+                return __get_response_dict(status = StatusCodeEnum.SUCCESS, result = {'address': address}) 
+            else:
+                return __get_response_dict(status = StatusCodeEnum.FAILED, error = "address not found")
+            
+    except Exception as e:
+        return __get_response_dict(status = StatusCodeEnum.FAILED, error = str(e))
+
+def zcl_add_network(node_id: int, ssid: str, password: str, endpoint_id: Optional[int] = 1, group_id: Optional[int] = 0, breadcrumb: Optional[int] = 0, timeoutMs: Optional[int] = 1000) -> Dict[str, Any] :
+    try:
+        __check_supported_os()
+        args = {}
+        args['ssid'] = ssid.encode("utf-8") + b'\x00'
+        args['credentials'] = password.encode("utf-8") + b'\x00'
+        args['breadcrumb'] = breadcrumb
+        args['timeoutMs'] = timeoutMs 
+        err, res = device_manager.devCtrl.ZCLSend("NetworkCommissioning", "AddWiFiNetwork", node_id, endpoint_id, group_id, args, blocking=True)
+        if err != 0:
+            return __get_response_dict(status = StatusCodeEnum.FAILED)
+        elif res != None:
+            return __get_response_dict(status = StatusCodeEnum.SUCCESS, result = str(res))
+        else:
+            return __get_response_dict(status = StatusCodeEnum.SUCCESS)
+
+    except Exception as e:
+        return __get_response_dict(status = StatusCodeEnum.FAILED, error = str(e))
+
+def zcl_enable_network(node_id: int, ssid:str, endpoint_id: Optional[int] = 1, group_id: Optional[int] = 0, breadcrumb: Optional[int] = 0, timeoutMs: Optional[int] = 1000) -> Dict[str, Any]:
```

Comment:
Would it be easier to expose the `ZCLSend` api directly, so we don't need to add all the APIs in the RPC, but define them in our test harness?

I think that would be a bit easier to maintain, but maybe I am missing something.

Reply:
This was discussed with @wahajsyed . We can certainly add zclSend in future. Currently we have added basic commands and see in next iteration if `zclSend` makes sense or not.